education_evaluation_notebook/models/education_notebook_line.py

```diff
@@ -0,0 +1,178 @@
+    eval_type = fields.Selection(selection=[
+        ('first', 'First'),
+        ('second', 'Second'),
+        ('third', 'Third'),
+        ('final', 'Final')],
+        string='Evaluation season', default='final', required=True
+    )
+    exam_ids = fields.One2many(comodel_name='education.exam',
+                               inverse_name='n_line_id', string='Exams',
+                               editable=True)
+    num_exams = fields.Integer(compute='_compute_num_exams',
+                               string='Number of exams')
+    competence_type_id = fields.Many2one(comodel_name='education.competence.type',
+                                         string='Competence type')
+    father_competence_id = fields.Many2one(comodel_name='education.notebook.line',
+                                         string='Father competence (notebobok line)',
+                                         store=True)
+    exists_master = fields.Boolean(string='Is master', compute='_compute_master_competences',
+                                   default=False)
+    #Se deberia de haber llamado notebook_line_for_father_competence
+    competence_son_ids = fields.One2many(comodel_name='education.notebook.line',
+                                         inverse_name='id',
+                                         string='Son competences (notebook lines)',
+                                         compute='populate_competence_child_list',
+                                         #domain=[('father_competence_id.id','=', id)],
+                                         editable=True)
+    father_comp_eval_check = fields.Boolean(string="Evaluation check for the father competence",
+                                            related='competence_id.evaluation_check',
+                                            comodel_name='education.competence',
+                                            default=False)
+    father_comp_global_check = fields.Boolean(string="Global check for the father competence",
+                                              related='competence_id.global_check',
+                                              comodel_name='education.competence',
+                                              default=False)
+    exams_academic_records_count = fields.Integer(compute='_compute_exams_academic_records_count',
+                                        string='Academic records of exams')
+    #academic_record_ids = fields.One2many(comodel_name='education.record',
+    #                                     inverse_name='id',
+    #                                     compute='populate_academic_records',
+    #                                     string='Academic records',
+    #                                     editable=True)
+
+    #COMPRUEBA LA JERARUIA DE LAS COMPETENCIAS
+    @api.depends('competence_id','father_competence_id')
+    @api.constrains('father_competence_id')
+    @api.one
```

Comment:
Has puesto `@api.one` y luego has añadido un for, eso no es correcto, aparte de que no se pone `@api.one`